src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -19,21 +18,18 @@ namespace NuGet.DependencyResolver
     public static class ResolverUtility
     {
         public static Task<GraphItem<RemoteResolveResult>> FindLibraryCachedAsync(
-            ConcurrentDictionary<LibraryRangeCacheKey, Task<GraphItem<RemoteResolveResult>>> cache,
             LibraryRange libraryRange,
             NuGetFramework framework,
             string runtimeIdentifier,
             RemoteWalkContext context,
             CancellationToken cancellationToken)
         {
-            var key = new LibraryRangeCacheKey(libraryRange, framework);
+            LibraryRangeCacheKey key = new(libraryRange, framework);
 
-            if (cache.TryGetValue(key, out var graphItem))
-                return graphItem;
-
-            graphItem = cache.GetOrAdd(key, FindLibraryEntryAsync(key.LibraryRange, framework, runtimeIdentifier, context, cancellationToken));
-
-            return graphItem;
+            return context.FindLibraryEntryCache.GetOrAddAsync(key,
+                state => FindLibraryEntryAsync(state.LibraryRange, state.framework, state.runtimeIdentifier, state.context, state.cancellationToken),
```

Comment:
Can this be marked `static`? This helps catch cases where there's an unintended capture.